config/config/src/main/java/io/helidon/config/MetaProviders.java

```diff
@@ -0,0 +1,273 @@
+import io.helidon.config.internal.UrlOverrideSource;
+import io.helidon.config.spi.ConfigSource;
+import io.helidon.config.spi.ConfigSourceProvider;
+import io.helidon.config.spi.OverrideSource;
+import io.helidon.config.spi.OverrideSourceProvider;
+import io.helidon.config.spi.PollingStrategy;
+import io.helidon.config.spi.PollingStrategyProvider;
+import io.helidon.config.spi.RetryPolicy;
+import io.helidon.config.spi.RetryPolicyProvider;
+
+/**
+ * Access to Java service loaders for config sources, retry policies and polling strategies.
+ */
+final class MetaProviders {
+    private static final List<ConfigSourceProvider> CONFIG_SOURCE_PROVIDERS;
+    private static final List<RetryPolicyProvider> RETRY_POLICY_PROVIDERS;
+    private static final List<PollingStrategyProvider> POLLING_STRATEGY_PROVIDERS;
+    private static final List<OverrideSourceProvider> OVERRIDE_SOURCE_PROVIDERS;
+
+    private static final Set<String> SUPPORTED_CONFIG_SOURCES = new HashSet<>();
+    private static final Set<String> SUPPORTED_RETRY_POLICIES = new HashSet<>();
+    private static final Set<String> SUPPORTED_POLLING_STRATEGIES = new HashSet<>();
+    private static final Set<String> SUPPORTED_OVERRIDE_SOURCES = new HashSet<>();
+
+    static {
+        CONFIG_SOURCE_PROVIDERS = HelidonServiceLoader
+                .builder(ServiceLoader.load(ConfigSourceProvider.class))
+                .addService(new BuiltInConfigSourcesProvider())
+                .build()
+                .asList();
+
+        CONFIG_SOURCE_PROVIDERS.stream()
+                .map(ConfigSourceProvider::supported)
+                .forEach(SUPPORTED_CONFIG_SOURCES::addAll);
+
+        RETRY_POLICY_PROVIDERS = HelidonServiceLoader
+                .builder(ServiceLoader.load(RetryPolicyProvider.class))
+                .addService(new BuiltInRetryPolicyProvider())
+                .build()
+                .asList();
+
+        RETRY_POLICY_PROVIDERS.stream()
+                .map(RetryPolicyProvider::supported)
+                .forEach(SUPPORTED_RETRY_POLICIES::addAll);
+
+        POLLING_STRATEGY_PROVIDERS = HelidonServiceLoader
+                .builder(ServiceLoader.load(PollingStrategyProvider.class))
+                .addService(new BuiltInPollingStrategyProvider())
+                .build()
+                .asList();
+
+        POLLING_STRATEGY_PROVIDERS.stream()
+                .map(PollingStrategyProvider::supported)
+                .forEach(SUPPORTED_POLLING_STRATEGIES::addAll);
+
+        OVERRIDE_SOURCE_PROVIDERS = HelidonServiceLoader
+                .builder(ServiceLoader.load(OverrideSourceProvider.class))
+                .addService(new BuiltinOverrideSourceProvider())
+                .build()
+                .asList();
+
+        OVERRIDE_SOURCE_PROVIDERS.stream()
+                .map(OverrideSourceProvider::supported)
+                .forEach(SUPPORTED_OVERRIDE_SOURCES::addAll);
+    }
+
+    private MetaProviders() {
```

Comment:
In case of util classes there should also be an exception fired if this constructor is call by reflection or some other way.

Reply:
Not really. It is an option. If we do it, we should have a test that validates that. So we do not do it.